emcc.py

```diff
@@ -1990,8 +1991,13 @@ def compile_source_file(i, input_file):
     link_flags = [l for l in link_flags if l[1] not in ('-lm', '-lrt', '-ldl', '-lpthread')]
     temp_files = filter_out_dynamic_libs(temp_files)
 
-    # Decide what we will link
-    if not (shared.Settings.WASM_BACKEND and shared.Settings.WASM_OBJECT_FILES):
+    link_to_object = final_suffix not in executable_endings
+    using_lld = shared.Settings.WASM_BACKEND and not (link_to_object and not shared.Settings.WASM_OBJECT_FILES)
+    if using_lld:
+      # Filter out link flags that lld doesn't support.  bind_at_load is often
+      # passed on OSX because libtool/autoconf add this link flag.
+      link_flags = [f for f in link_flags if f[1] not in ('-bind_at_load',)]
```

Comment:
Should we filter `-rpath` as well to solve the Rust issue? Alternatively we can debug Rust and figure out why it's passing rpath when it's not supposed to be doing so.

Reply:
Good idea!